src/main/java/cn/nukkit/Server.java

```diff
@@ -1483,11 +1486,11 @@ private void saveOfflinePlayerDataInternal(PlayerDataSerializer serializer, Comp
 
     private void convertLegacyPlayerData() {
         File dataDirectory = new File(getDataPath(), "players/");
-        Pattern uuidPattern = Pattern.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}.dat$");
 
         File[] files = dataDirectory.listFiles(file -> {
             String name = file.getName();
-            return !uuidPattern.matcher(name).matches() && name.endsWith(".dat");
+            Matcher matcher = UUID_PATTERN.matcher(name);
```

Comment:
local variable is redundant

Reply:
This solution improves readability. As for me, this solution will not affect the literacy and performance of this solution.

Reply:
fine, i guess that *is* just a style thing. we really need to establish a standard coding style...